aten/src/ATen/nn_parse.py

```diff
@@ -245,8 +245,7 @@ def function_info(name, arguments, cimpls, buffers, backends, inplace, scalar_ch
         'variants': ['function'],
     }
 
-
-def base_declaration(func, thnn_function, backends, inplace=False):
+def base_declaration(func, thnn_function, backends, backend_types, inplace=False):
```

Comment:
what is the difference between `backends` and `backend_types`?

Can you just call the scalar types "dtypes"?

Reply:
It's a map of backends and corresponding backend types. https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/preprocess_declarations.py#L34